sklearn/utils/_array_api.py

```diff
@@ -110,7 +110,7 @@ def size(x):
 
 def _is_numpy_namespace(xp):
     """Return True if xp is backed by NumPy."""
-    return xp.__name__ in {"numpy", "array_api_compat.numpy", "numpy.array_api"}
+    return xp.__name__ in {"numpy", "array_api_compat.numpy"}
```

Comment:
I like this change: `_is_numpy_namespace` now really a marker for the NumPy case and `array-api-strict` is now just considered like any other non-NumPy array api implementation only to be used for testing purposes and therefore not being subject to any special treatment motivated by backward-compat considerations.